engine/common/synchronization/engine.go

```diff
@@ -178,6 +178,33 @@ func (e *Engine) setupResponseMessageHandler() error {
 				}
 				return ok
 			},
+			Map: func(msg *engine.Message) (*engine.Message, bool) {
+				blockResponse, ok := msg.Payload.(*messages.BlockResponse)
+				if !ok {
+					// should never happen, unless there is a bug.
+					e.log.Fatal().
+						Hex("origin_id", logging.ID(msg.OriginID)).
+						Interface("payload", msg.Payload).
+						Msg("cannot match the payload to BlockResponse")
+					return nil, false
+				}
+				proposals, err := blockResponse.BlocksInternal()
+				if err != nil {
+					e.log.Warn().
```

Comment:
Can we add TODO specifically for replacing this log with call to a protocol violation consumer?